tools/logger/CMakeLists.txt

```diff
@@ -3,6 +3,7 @@
 add_executable(sof-logger
 	logger.c
 	convert.c
+	filter.c
```

Comment:
So my understanding is

1) create UUID filter text file.
2) pass UUID fillet to logger ?

This seems overly complex and cumbersome. UUID is also not the instance ID, meaning we enable all UUIDs for typeX and not just the one we care about

We should really be filtering on component instance name. e.g. BUF1.1, VOL2..0 etc and do it via the command line
```sof-logger -fdebug="BUF1.1, VOL1.0" -finfo="BASEFW" -fvdebug="SRC1.1"```

We should also be able to select the pipeline too

```sof-logger -fdebug="PIPELINE1"```

Reply:
What's a use case for such a trace logger granuality? I guess it's rather error prone solution, because of trace filtering parameters incompatibility between different platforms/topologies.

Reply:
Also when we would like to have verbose logs from many components, then cmd line to run might be quite long and unreadable. Moreover file usage, guarantee the same configuration between debug sessions, what may be not true for command line arguments.

Reply:
Can't unrecognized parameters simply be ignored, perhaps with a warning in the traces?

Reply:
I rather thought about situation when given parameters are valid, but between different topology refer to differ components (in sense of usage, not type), so it may be unintended change.

Reply:
@ktrzcinx use case is that I want to trace pipeline 0 (comps X, Y) , and not the same component type on other (connected or otherwise) pipelines. I can also read the topology (or view the topology image) and know the strings I need to pass without have to manually convert topology elements to UUIDs

Please also state why you think it's error prone ? Copying a 32 byte ASCII is UUID is error prone.

Command line usage is exactly the same as UUID file usage. It can be scripted or sent via email if needed.

Reply:
I think about such a situation - we have topology 1 like:
`BUF1.1 -> VOL1.1 -> BUF1.2 -> SmartAmp1.2 -> BUF1.3 -> DAI1.3`
`BUF2.1 -> FIR2.1 -> BUF2.2 -> SmartAmp2.2 -> BUF2.3 -> DAI2.3`
and we found some issue on SmartAmp2.2, what is seen only in verbose logs for SmartAmp connected after FIR component what isn't known yet (hypothetical situation). After try of reproduction on another platform we didn't see any troubles, because topology 2 is:
`BUF1.1 -> FIR1.1 -> BUF1.2 -> SmartAmp1.2 -> BUF1.3 -> DAI1.3`
`BUF2.1 -> VOL2.1 -> BUF2.2 -> SmartAmp2.2 -> BUF2.3 -> DAI2.3`

*) Component connections layout is only for example purpose
  
Here, different tracking levels for different `SmartAmp` hide error, but filtering parameters are valid for both topology. 

32 byte ASCII can be easily validated, readed from source code, output logs, ldc file dump or generated filter template config file. Moreover UUID doesn't change at all, so basically we need to check it once, in opposite to component instance name which one need to be check for each topology individually.

Reply:
@ktrzcinx sorry not getting your point here. If I want to debug pipeline 1 I want to

```sof-logger -fdebug="pipeline1"```

Thats all. Likewise if I want to debug all smart amps I want to

```sof-logger -fdebug="SmartAmp*"```

My key point is that the debugging must be really simple and intuative. e.g. I want to enable debug for everything

```sof-logger --fdebug="*"```

or I want to debug pipeline2 except for the DAI (as it can spam my logs)

```sof-logger -fdebug="pipeline2, !DAI2.3```

or I want to debug the allocator and FIRs

```sof-logger -fdebug="sys.alloc, FIR*"```

Reply:
> `sof-logger -fdebug="pipeline2, !DAI2.3`

Perhaps following this convention, such an example also should be valid:
`sof-logger -fdebug="pipeline2, !DAI2.3, !DAI2.1`
What results in quite complicated messages parsing and handling. When we little change convention, and do it like this:
`sof-logger -fdebug="pipeline2" -ferror="DAI2.3, DAI2.1"`
Sending messages from left to right, so following config can overwrite previous one, then implementation will be much easier to implement and maintain.

Reply:
@ktrzcinx yep, both examples you gave there would be valid. The last passed argument overrides previous args (this makes the parsing simple). I would imagine you would have a list of all objects (with a trace level variable) and set the level for each object as the args were parsed. Yes, so previous args could be overridden, but that's OK and expected (i.e. GCC does this).
Do you need an IPC from the kernel to get all the valid objects or can you read all the data you need today via sysFS ? If you need a new IPC then it should export all the data you need via sysfs. i.e. reading the sysfs file would send the IPC to FW.
Btw, we would also need to standardise the naming of the basefw "system" components, maybe we could have a "sys." prefix i.e. "sys.allocator", "sys.scheduler.edf" and so on (we can then use "sys.*" as an arg.